blocks/rich-text/index.js

```diff
@@ -418,6 +436,50 @@ export class RichText extends Component {
 		};
 	}
 
+	setInsertPosition() {
+		const container = this.getContainerNode();
+		const containerStyle = window.getComputedStyle( container );
+		const marginLeft = get( containerStyle, 'margin-left', 0 );
+		const containerPosition = container.getBoundingClientRect();
+		const rect = getRectangleFromRange( this.editor.selection.getRng() );
+		const insertPosition = {
+			top: rect.top - containerPosition.top,
+			left: rect.right - containerPosition.left,
+			marginLeft,
+			height: rect.height,
+		};
+
+		this.setState( { insertPosition } );
+	}
+
+	insertInlineBlock() {
+		const { inlineBlockForInsert, completeInlineInsert } = this.props;
+
+		if ( inlineBlockForInsert.type === 'image' ) {
+			this.setState( { mediaLibraryOpen: true } );
+		} else {
+			inlineBlockForInsert.render( this.editor );
```

Comment:
Is there any way we can avoid `toggleInsertAvailable` by doing some slot/fill magic? Maybe this is not possible, just wondering out loud. :)